### PR TITLE
PMM-9729 update go to 1.18

### PR DESCRIPTION
**Pull request by artemgavrilov**
PMM-9729

Build: SUBMODULES-0

- [ ] Links to other linked pull requests (optional).

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -56,5 +56,3 @@ jobs:
           make test-crosscover
           make test-race
 
-      - name: Run required checkers and linters.
-        run: make check
```
Comment (artemgavrilov) [inline .github/workflows/go.yml:60]: We invoke `ci-reviewdog` few lines above

File: Makefile [hunk 2]
```diff
@@ -55,18 +53,15 @@ test-crosscover:                ## Run tests and collect cross-package coverage
 
 check:                          ## Run required checkers and linters.
 	go run .github/check-license.go
-
-check-style:                    ## Run style checkers and linters.
 	$(BIN_PATH)/golangci-lint run -c=.golangci.yml ./... --new-from-rev=main
 	$(BIN_PATH)/go-consistent -pedantic ./...
 
-check-all: check check-style    ## Run all linters for new code..
```
Comment (artemgavrilov) [inline Makefile:63]: Left single `check` target to have it consistent with other repos

File: Makefile [hunk 3]
```diff
@@ -55,18 +53,15 @@ test-crosscover:                ## Run tests and collect cross-package coverage
...
-
 FILES = $(shell find . -type f -name '*.go')
 
 format:                         ## Format source code.
-	gofmt -w -s $(FILES)
-	$(BIN_PATH)/goimports -local github.com/percona/pmm-admin -l -w $(FILES)
+	$(BIN_PATH)/gofumpt -l -w $(FILES)
+	$(BIN_PATH)/goimports -local github.com/percona/pmm-agent -l -w $(FILES)
```
Comment (percona-csalguero) [inline Makefile]: This `pmm-agent` is correct?

**Review by BupycHuk — APPROVED** [review]

**Review by percona-csalguero — CHANGES_REQUESTED** [review]

File: Makefile (hunk 3 above)
Comment (artemgavrilov) [reply Makefile]: Nope, copy-paste error. Fixed.

**Review by percona-csalguero — APPROVED** [review]

**Pull request merged**